src/main/java/ai/elimu/rest/v2/crowdsource/WordContributionRestController.java

```diff
@@ -103,45 +95,50 @@ public String handleWordContributionRequest(
-            response.setStatus(HttpStatus.BAD_REQUEST.value());
+            response.setStatus(HttpStatus.CONFLICT.value());
 
             String jsonResponse = jsonObject.toString();
             logger.info("jsonResponse: " + jsonResponse);
             return jsonResponse;
         }
 
         try {
-            //Convert the JSON String to WordGson Object.
-            Word newWord = new Word();
-            newWord.setWordType(wordGson.getWordType());
-            newWord.setText(wordGson.getText());
-            List<LetterToAllophoneMappingGson> wordGsonLetterToAllophoneMappingsGson = wordGson.getLetterToAllophoneMappings();
+            // Convert the WordGson to Word POJO.
+            Word word = new Word();
+            word.setWordType(wordGson.getWordType());
+            word.setText(wordGson.getText());
+            List<LetterToAllophoneMappingGson> letterToAllophoneMappingsGsons = wordGson.getLetterToAllophoneMappings();
             List<LetterToAllophoneMapping> letterToAllophoneMappings = new ArrayList<>();
-            for (LetterToAllophoneMappingGson letterToAllophoneMappingGson : wordGsonLetterToAllophoneMappingsGson) {
-                LetterToAllophoneMapping letterToAllophoneMapping = letterToAllophoneMappingDao.read(letterToAllophoneMappingGson.getId());
+            for (LetterToAllophoneMappingGson letterToAllophoneMappingGson : letterToAllophoneMappingsGsons) {
+                LetterToAllophoneMapping letterToAllophoneMapping =
+                        letterToAllophoneMappingDao.read(letterToAllophoneMappingGson.getId());
                 letterToAllophoneMappings.add(letterToAllophoneMapping);
             }
-            newWord.setLetterToAllophoneMappings(letterToAllophoneMappings);
-            wordDao.create(newWord);
-
-            //Convert Request Body(String) to JSON Object.
-            JSONObject requestJson = new JSONObject(requestBody);
+            word.setLetterToAllophoneMappings(letterToAllophoneMappings);
+            wordDao.create(word);
 
             WordContributionEvent wordContributionEvent = new WordContributionEvent();
             wordContributionEvent.setContributor(contributor);
             wordContributionEvent.setTime(Calendar.getInstance());
-            wordContributionEvent.setWord(newWord);
-            wordContributionEvent.setRevisionNumber(newWord.getRevisionNumber());
-            wordContributionEvent.setComment(requestJson.getString("contributionComment"));
-            wordContributionEvent.setTimeSpentMs(System.currentTimeMillis() - Long.valueOf(requestJson.getString("timeStart")));
+            wordContributionEvent.setWord(word);
+            wordContributionEvent.setRevisionNumber(word.getRevisionNumber());
+            wordContributionEvent.setComment(wordContributionEvent.getComment());
+            wordContributionEvent.setTimeSpentMs(System.currentTimeMillis() -
+                    wordContributionEvent.getTime().getTimeInMillis());
```

Comment:
Actually, this piece of information refers to the time passed during the creation/editing of the Word, which is time that would pass while the contributor was interacting with the Crowdsource app itself. However, I see that the field `timeSpentMs` is missing from the `model` library, so your implementation is a good temporary solution until we add the missing property. I added an issue for it here: https://github.com/elimu-ai/model/projects/7#card-62107646

```suggestion
            // TODO: wordContributionEvent.setTimeSpentMs(wordContributionEventGson.getTimeSpentMs());
            wordContributionEvent.setTimeSpentMs(System.currentTimeMillis() -
                    wordContributionEvent.getTime().getTimeInMillis());
```